src/com/company/balance/GiftCardBalance.java

```diff
@@ -12,9 +13,15 @@ public GiftCardBalance(UUID customerId, Double balance) {
     }
 
     @Override
-    public Double addBalance(Double additionalBalance) {
-        double promotionAmount = additionalBalance * 10 / 100;
-        setBalance(getBalance() + additionalBalance + promotionAmount);
+    public Double addBalance(int selection) {
+        if (selection==2){
+            System.out.println("How much you would like to add?");
+            double additionalAmount = new Scanner(System.in).nextDouble();
+            double promotionAmount = additionalAmount * 10 / 100;
+            setBalance(getBalance() + additionalAmount + promotionAmount);
+            System.out.println("New giftCard balance: " + getBalance());
+        }
```

Comment:
This part is breaking "Single Responsibility" rule of SOLID.
Because, printing something on the console and getting user input is not this class' responsibility.
Please revert this change not to brake the main flow

Reply:
done